lib/puppet/provider/exec/powershell.rb

```diff
@@ -45,9 +45,13 @@ def self.upgrade_message
     @upgrade_warning_issued = true
   end
 
-  def ps_manager
+  def ps_manager(resource)
     debug_output = Puppet::Util::Log.level == :debug
-    Pwsh::Manager.instance(Pwsh::Manager.powershell_path, Pwsh::Manager.powershell_args, debug: debug_output)
+    if resource[:timeout].nil?
+      Pwsh::Manager.instance(Pwsh::Manager.powershell_path, Pwsh::Manager.powershell_args, debug: debug_output)
+    else
+      Pwsh::Manager.instance(Pwsh::Manager.powershell_path, Pwsh::Manager.powershell_args, debug: debug_output, pipe_timeout: resource[:timeout])
```

Comment:
IIRC pipe_timeout can be nil and ruby_pwsh uses it default.  So probbaly no need for the if?

Reply:
Actually do we need the whole resource? could it just be the timeout value?